crates/cheatnet/src/forking/worker.rs

```diff
@@ -0,0 +1,143 @@
+use crate::conversions::{class_hash_to_felt, contract_address_to_felt};
+use blockifier::execution::contract_class::{
+    ContractClass as ContractClassBlockifier, ContractClassV1,
+};
+use blockifier::execution::execution_utils::stark_felt_to_felt;
+use blockifier::state::errors::StateError::StateReadError;
+use blockifier::state::state_api::StateResult;
+use cairo_lang_starknet::abi::Contract;
+use cairo_lang_starknet::casm_contract_class::CasmContractClass;
+use cairo_lang_starknet::contract_class::{ContractClass, ContractEntryPoints};
+use cairo_lang_utils::bigint::BigUintAsHex;
+use num_bigint::BigUint;
+use starknet::core::types::{BlockId, ContractClass as ContractClassStarknet, FieldElement};
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::{JsonRpcClient, Provider};
+use starknet_api::core::{ClassHash, ContractAddress, Nonce};
+use starknet_api::hash::{StarkFelt, StarkHash};
+use starknet_api::state::StorageKey;
+use tokio::runtime::Runtime;
+use url::Url;
+
+#[derive(Debug)]
+pub struct Worker {
+    client: JsonRpcClient<HttpTransport>,
+    block_id: BlockId,
+}
+
+impl Worker {
+    #[must_use]
+    pub fn new(url: &str, block_id: BlockId) -> Self {
+        Worker {
```

Comment:
It would be probably better to keep one runtime, as afaik creating new runtime spawns a thread each time.